xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -4020,7 +4020,7 @@ dht_file_counter_thread(void *args)
     dht_build_root_loc(defrag->root_inode, &root_loc);
 
     while (defrag->defrag_status == GF_DEFRAG_STATUS_STARTED) {
-        timespec_now(&time_to_wait);
```

Comment:
Shouldn't this entire code block move outside of the while loop (a couple of lines up)? I assume we want to wait 10 minutes only once.
A similar code block should exist inside the loop as well, but with a much shorter wait time, 15 seconds sounds good?

Reply:
The original timeout is 10min. I agree we can reduce it after the 1st 10min, but not sure regarding the 15sec.
The thread calls syncop_statfs for each subvols (which btw could be changed to a local call) and do more stuff.
My suggestion is to merge this fix to prevent the thread from repetitive work and log.
I"ll open ER for checking thread works, which may be optimized and for defining a new timeout. 
@BarakSason - what do you think?

Reply:
Sounds acceptable to me, as this fix solves the problem of the thread spinning endless in a loop, wasting resources and spams the log file, and solving this is the priority.
Please do create an issue for the other changes mentioned.